firmware/pico/src/app/hal.cc

```diff
@@ -1,8 +1,51 @@
 #include "hal.hh"
 
+#include <hardware/exception.h>
+#include <hardware/structs/systick.h>
+
 #include "hardware/spi.h"
 #include "pico/stdlib.h"
 
 uint64_t get_time_since_boot_us() { return to_us_since_boot(get_absolute_time()); }
 
-uint32_t get_time_since_boot_ms() { return to_ms_since_boot(get_absolute_time()); }
\ No newline at end of file
+uint32_t get_time_since_boot_ms() { return to_ms_since_boot(get_absolute_time()); }
+
+volatile uint32_t ClockSource::_interruptCount __attribute__((aligned(4)));
+
+ClockSource::ClockSource() {
+    /* register description: ArmV6-M Manual B3.3 */
+
+    systick_hw->csr = 0 |        /* Enable: false            */
+                      (1 << 1) | /* Interrupt: true          */
+                      (1 << 2);  /* Clock source: internal   */
+
+    systick_hw->rvr = 0x00FFFFFF; /* Max counter size (24 bit)*/
+    systick_hw->cvr = 0;          /* Clear current value */
+
+    exception_set_exclusive_handler(SYSTICK_EXCEPTION, handleIrq);
+
+    systick_hw->csr = systick_hw->csr | 1; /* Start timer */
+}
+
+ClockSource::~ClockSource() { systick_hw->csr = 0; }
+
+ClockSource& ClockSource::instance() {
+    static ClockSource inst;
+    return inst;
+}
+
+uint64_t ClockSource::get48MHzTickCount() {
```

Comment:
Please add doxygen style comments in the header file so that others can understand how the counter works and what it's used for! The old comments from the ADSBee member functions that got removed have most of the important info that could be reused.